File: array2.pas
program array2;
uses crt;

type
M = array[1..5,1..5] of integer;
var
m1,m2,m3: M;
x,y,i,j,x1,x2,y1,y2,a,b : integer;

begin
clrscr;
writeln;
writeln;

writeln('Silahkan masukkan jumlah kolom dan baris matriks anda');
writeln;
write('Masukkan jumlah baris matriks ke-1: ');readln(x1);
write('Masukkan jumlah kolom matriks ke-1: ');readln(y1);
write('Masukkan jumlah baris matriks ke-2: ');readln(x2);
write('Masukkan jumlah kolom matriks ke-2: ');readln(y2);

clrscr;

    if (x1=x2) and (y1=y2) and (x1<=5) and (y1<=5) then
     begin

        for i := 1 to x1 do
            begin
                for j := 1 to y1 do
                 begin
                    write('Masukkan elemen Matriks 1 [',i,',',j,']: ');readln(m1[i,j]);
                 end;
            end;

    writeln;

        for a := 1 to x2 do
            begin
                for b := 1 to y2 do
                 begin 
                    write('Masukkan elemen Matriks 2 [',a,',',b,']: ');readln(m2[a,b]);
                 end;
            end;


        for x:= 1 to x1 do
            begin  
                for y:= 1 to y1 do
                 begin 
                    m3[x,y] := m1[i,j] + m2[a,b];
                 end;
            end;

        writeln('Hasil penjumlahan matriks: ');
        for x:= 1 to x1 do
            begin   
                for y:= 1 to y2 do 
                 begin
                    write(m3[x,y]:4,' ');
                 end;
            end;

        writeln;
        end

    else if (x1>5) or (y1>5) then
        begin
         writeln('Maksimum kolom dan baris adalah 5!')
        end

    else
        begin
         writeln('Matriks tidak dapat di jumlahkan karena berbeda ordo!');
        end;
    readln;
end.

File: Array2.py
import os

os.system("cls")

import numpy as np

print("Masukkan ordo matriks! (maks 5x5) ")
print()

n = int(input("Masukkan jumlah Baris matriks-1: "))
m = int(input("Masukkan jumlah Kolom matriks-1: "))
o = int(input("Masukkan jumlah Baris matriks-2: "))
p = int(input("masukkan jumlah kolom matriks-2: "))

if (n == o ) & (m == p) & (m <= 5) & (n <= 5):
    
    matrix1 = []
    matrix2 = []
    print()
    print("masukkan inputan!")
    print()

    for i in range (n):
        lst1 = []
        for j in range (m):
            print("Masukkan matriks 1: [",i+1,",",j+1,"]",end =" ")
            lst1.append(int(input()))
        matrix1.append(lst1)

    print()

    for i in range (n):
        lst2 = []
        for j in range (m):
            print("Masukkan matriks 2: [",i+1,",",j+1,"]",end =" ")
            lst2.append(int(input()))
        matrix2.append(lst2)



    print()

    for i in range (n):
        for j in range (m):
            print(matrix1[i][j],end =" ")
            print()

    
    print()

    for i in range (n):
        for j in range (m):
            print(matrix2[i][j],end =" ")
            print()
else:
    os.system("cls")
    print("Matriks  tidak dapat di jumlahkan")
